### PR TITLE
Add Narayana example

### PR DESCRIPTION
**Pull request by fvaleri**
This example demonstrates how to run a Camel Service on Spring Boot that supports JTA transactions on two external transactional resources: a database (MySQL) and a message broker (Artemis).

### CODE REVIEW

**Comment by fvaleri** [comment]
@grgrzybek @luigidemasi

**Review by oscerd — APPROVED** [review]

**Review by zhfeng — APPROVED** [review]

File: spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        rest("/messages")
+            .produces("text/plain")
+            .get()
+                .to("direct:messages")
+            .post("/{message}")
+                .param().name("message").type(RestParamType.path).dataType("string").endParam()
+                .to("direct:trans");
+
+        from("direct:messages")
+            .to("jpa:it.fvaleri.integ.AuditLog?namedQuery=getAuditLog")
```
Comment (davsclaus) [inline spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java:43]: The package name should likely be sample.camel

File: spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+@Component
+public class CamelRoutes extends RouteBuilder {
+    @Value("${camel.servlet.mapping.context-path}")
+    public String contextPath;
+
+    @Override
+    public void configure() {
+        restConfiguration()
+            .contextPath(contextPath.substring(0, contextPath.length() - 2));
```
Comment (davsclaus) [inline spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java:32]: Can you add code comment why you - 2

File: spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        from("direct:messages")
+            .to("jpa:it.fvaleri.integ.AuditLog?namedQuery=getAuditLog")
+            .convertBodyTo(String.class);
+
+        from("direct:trans")
+            .transacted()
+            .setBody(simple("${headers.message}"))
+            .to("bean:auditLog?method=createAuditLog(${body})")
+            .to("jpa:it.fvaleri.integ.AuditLog")
```
Comment (davsclaus) [inline spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java:50]: Same here

File: spring-boot-jta-jpa/src/main/resources/spring-camel.xml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                <!-- disable local transactions as JTA TM will take care of enrolling -->
+                <property name="transacted" value="false"/>
+                <!-- caching does not work with distributed transactions -->
+                <property name="cacheLevelName" value="CACHE_NONE"/>
+                <property name="maxConcurrentConsumers" value="1"/>
+                <property name="testConnectionOnStartup" value="true"/>
+            </bean>
+        </property>
+    </bean>
+</beans>
```
Comment (davsclaus) [inline spring-boot-jta-jpa/src/main/resources/spring-camel.xml:113]: Its a bit sad or scary that it takes 100+ lines of spring bean configuration to setup JTA with a database and JMS :(

File: spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java (hunk 1 above)
Comment (fvaleri) [reply spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java]: Yep, I forgot to rename it when copying from my personal repo. The funny thing is that it still works (bug or feature?). Anyway, thanks for spotting this.

File: spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java (hunk 2 above)
Comment (fvaleri) [reply spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java]: I took it from the `spring-boot` example, but it is not needed. Thanks.

File: spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java (hunk 3 above)
Comment (fvaleri) [reply spring-boot-jta-jpa/src/main/java/sample/camel/CamelRoutes.java]: Yep.

File: spring-boot-jta-jpa/src/main/resources/spring-camel.xml (hunk 4 above)
Comment (fvaleri) [reply spring-boot-jta-jpa/src/main/resources/spring-camel.xml:113]: Agreed. One of the objective of this example was to make that complexity explicit. It is even more complicated if you look at beans graph resulting from Narayana SB auto-configuration. There additional wrapper and proxy classes.

**Review by davsclaus — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

**Pull request merged**